cpp/src/arrow/acero/asof_join_node.cc

```diff
@@ -688,14 +825,19 @@ class InputState {
 
     // Not updated.  Try to update and possibly advance.
     bool advanced, updated = false;
+    OnType latest_time;
```

Comment:
Why this change?

Reply:
This just exposes `latest_time` outside the loop, for code just after it. Its type is unchanged.